### ISSUE TITLE
Unable to find template plugin matching 'typescript-operations'

### ISSUE THREAD

**Issue by carlosbaraza** [dotansimha/graphql-code-generator#6715]
**Describe the bug**
Same problem described in #2043 and #3769 and #6629

Using `typescript-operations` would throw the following error if `@babel/runtime` is not installed. Some plugins try to require `@babel/runtime/helpers/objectSpread2`, which cause the error: `Error: Cannot find module '@babel/runtime/helpers/objectSpread2'`
```
Unable to find template plugin matching 'typescript-operations'
```

**To Reproduce**
1. Uninstall `@babel/runtime`.
2. Use `typescript-operations`

package versions:
```
    "@graphql-codegen/cli": "^2.2.0",
    "@graphql-codegen/introspection": "^2.1.0",
    "@graphql-codegen/schema-ast": "^2.2.0",
    "@graphql-codegen/typescript": "^2.2.2",
    "@graphql-codegen/typescript-graphql-request": "^4.1.4",
    "@graphql-codegen/typescript-operations": "^2.1.4",
    "@graphql-codegen/typescript-react-apollo": "^3.1.4",
    "@graphql-codegen/typescript-resolvers": "^2.2.1",
```

**Expected behavior**
1. The error message should be more helpful. The [error handling here is catching every single `MODULE_NOT_FOUND` error](https://github.com/dotansimha/graphql-code-generator/blob/master/packages/graphql-codegen-cli/src/plugins.ts#L24-L35) instead of checking that the MODULE_NOT_FOUND is actually because of the actual module was not found, instead of a dependency of the plugin. 
2. `@babel/runtime` should be a dependency of the plugins if the use the package

**Comment by Fxlr8** [comment]
This is kinda silly, but running `npm install` after the graphql-codegen init process fixed the problem for me. Looks like the script adds those plugins to the `package.json`, but it does not install them.

**Comment by vinnymac** [comment]
Running `npm install` after using the `graphql-codegen` cli wizard did not fix this for me. I had to manually do `npm install -save-dev @babel/runtime` to work around this issue. Since I am using typescript without babel, this dependency was not part of my project.

Curious why it would fix it for some but not all.

**Comment by ardatan** [comment]
@vinnymac Could you share a reproduction on CodeSandbox or in a GitHub repo?

**Comment by tvvignesh** [comment]
When trying to install manually:
![image](https://user-images.githubusercontent.com/1165845/147878459-ca2c9b38-01c6-429a-8966-eb8317b99ee4.png)

When using `npx graphql-codegen init` as mentioned in docs here: https://www.graphql-code-generator.com/docs/getting-started/installation#initialization-wizard

![image](https://user-images.githubusercontent.com/1165845/147878891-74b4d165-8172-4842-a1d1-1427a14639dc.png)

Looked for `@babel/runtime` and was not able to find it in the deps

![image](https://user-images.githubusercontent.com/1165845/147878769-22d5938e-866e-4541-8851-37eca6418f5d.png)

![image](https://user-images.githubusercontent.com/1165845/147878820-8b4f417b-a77c-406c-9e01-157c7649cdc6.png)

After doing an `npm install` once codegen does an init:

![image](https://user-images.githubusercontent.com/1165845/147878846-732bb47a-e5cf-4e02-93c5-ecde3e6a80f6.png)

And everything works now:

![image](https://user-images.githubusercontent.com/1165845/147878932-3dbb0716-f8c8-4c17-a572-d869d411682b.png)

In summary, the issue here is that codegen init just adds the plugins to the `package.json` file but does not install it as mentioned by @Fxlr8 and hence users were facing the issue. Closing this.

**Comment by Urigo** [comment]
@tvvignesh @dotansimha should we mention it in the docs then?
Or change the behaviour to do npm install?

**Comment by vinnymac** [comment]
> @vinnymac Could you share a reproduction on CodeSandbox or in a GitHub repo?

Unfortunately this was in a closed source project. If it’s not easily reproducible, it’s likely not worth spending the time on.

I do recommend mentioning the need for an `npm install` in the docs, if the intention is for that to be a manual step in the process of setup. It’d be nice if you linked directly to this discussion as well. For my case, `npm install` after `npx graphql-codegen init` did nothing at all.

**Comment by tvvignesh** [comment]
@Urigo Sure. Will send a PR to add it to the docs

**Comment by lemiesz** [comment]
the plugins really should bundle `@babel/runtime` having to manually install it is not always an option, and its kind of a nightmare for me right now trying to get this to worked with unejected CRA in a monorepo
